repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -1947,7 +1945,57 @@ private void addHasLineage(AttributeMutationContext ctx, EntityMutationContext c
                 }
 
                 if (removeAttr) {
-                    vertex.removeProperty(HAS_LINEAGE);
+                    AtlasGraphUtilsV2.setEncodedProperty(vertex, HAS_LINEAGE, false);
+                }
+            }
+        }
+        RequestContext.get().endMetricRecord(metricRecorder);
+    }
+
+    public void removeHasLineageOnDelete(Collection<AtlasVertex> vertices) {
+        MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("removeHasLineageOnDelete");
```

Comment:
This method can be defined in deleteHandler and can be used inside resetHasLineage method as well. This will help in avoiding code duplication

Reply:
refactoring done to move method in deleteHandler and reusing the code.